azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/util/AzureCliUtils.java

```diff
@@ -116,106 +68,49 @@ private static int compareWithMinimVersion(String version) {
-                switcher = WINDOWS_SWITCHER;
-            } else {
-                starter = LINUX_MAC_STARTER;
-                switcher = LINUX_MAC_SWITCHER;
-            }
-
-            final String commandWithMacPathFix = isWindows ? command : String.format("export PATH=$PATH:/usr/local/bin ; %s", command);
-            final ProcessBuilder builder = new ProcessBuilder(starter, switcher, commandWithMacPathFix);
-            final String workingDirectory = getSafeWorkingDirectory();
-            if (workingDirectory != null) {
-                builder.directory(new File(workingDirectory));
-            } else {
-                throw new IllegalStateException("A Safe Working directory could not be found to execute command from.");
-            }
-            builder.redirectErrorStream(true);
-            final Process process = builder.start();
-            reader = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8));
-            String line;
-            final StringBuilder output = new StringBuilder();
-            while (true) {
-                line = reader.readLine();
-                if (line == null) {
-                    break;
-                }
-                if (line.startsWith(WINDOWS_PROCESS_ERROR_MESSAGE) || line.matches(LINUX_MAC_PROCESS_ERROR_MESSAGE)) {
-                    throw new CredentialUnavailableException(
-                            "AzureCliTenantCredential authentication unavailable. Azure CLI not installed.");
-                }
-                output.append(line);
-            }
+            return CommandUtils.exec(command);
+        } catch (IOException e) {
+            throw new AzureToolkitAuthenticationException(
+                    String.format("Cannot execute azure cli command '%s' due to error: %s.", command, e.getMessage()));
+        }
+    }
 
-            final String processOutput = StringUtils.replace(output.toString(),
-                    "Argument '--tenant' is in preview. It may be changed/removed in a future release.", "");
-            process.waitFor(10, TimeUnit.SECONDS);
-            if (process.exitValue() != 0) {
-                if (processOutput.length() > 0) {
-                    final String redactedOutput = redactInfo(processOutput);
-                    if (redactedOutput.contains("az login") || redactedOutput.contains("az account set")) {
-                        throw new CredentialUnavailableException(
-                                "AzureCliTenantCredential authentication unavailable. Please run 'az login' to set up account. " +
-                                        "Detailed error is: " + processOutput);
-                    }
-                    throw new ClientAuthenticationException(redactedOutput, null);
-                } else {
-                    throw new ClientAuthenticationException("Failed to invoke Azure CLI ", null);
-                }
+    private static int compareWithMinimVersion(String version) {
```

Comment:
there are some existing wheels to compare versions
https://www.baeldung.com/java-comparing-versions